### DISCUSSION TITLE
Can I use gatsby-plugin-image in a paragraph ?

### DISCUSSION THREAD

**Discussion by Atsushi-Haba** [gatsbyjs/gatsby#33675] [Help]
Hello.
In normal React, the following will work
```
export const foo = () => {
  return (
    <p>
      <img src="/path/to/image" />
    </p>
  )
}
```

Using gatsby-plugin-image, the code should look like this
```
export const bar = () => {
  return (
    <p>
      <StaticImage src="/path/to/image" />
    </p>
  )
}
```
However, the html built with this is invalid, because the \<div\> is inside the \<p\>!
```
<p>
  <div data-gatsby-image-wrapper>
    <div ...>
      <img ...>
    </div>
    <div data-placeholder-image></div>
    <picture>...</picture>
    <noscript>...</noscript>
    <script>...</script>
  </div>
</p>
```

I read the documentation of gatsby-plugin-image and found the "as" props
```
export const foo = () => {
  return (
    <p>
      <StaticImage src="/path/to/image" as="span" />
    </p>
  )
}
```
But this doesn't work because it only changes the outer wrapper, the inner \<div\> is still there!
```
<p>
  <span data-gatsby-image-wrapper>
    <div ...>
      <img ...>
    </div>
    <div data-placeholder-image></div>
    <picture>...</picture>
    <noscript>...</noscript>
    <script>...</script>
  </span>
</p>
```

Is it possible to use gatsby-plugin-image in this way?

**Comment by LekoArts** [comment]
Hi!

No, you won't be able to achieve what you want. You'll need to place it outside of the `<p>`

**Reply by Atsushi-Haba** [reply]
Thank you!
So you are saying that gatsby-plugin-image can only be used inside html tags that can have Flow content(\<div\>) as children.

If I want to use an image inside a tag that only allows Phrasing content (such as \<p\> or \<h1\>), is it best to use publicURL and \<img\>?
